src/qengine/opencl.cpp

```diff
@@ -1132,14 +1176,13 @@ void QEngineOCL::ProbMaskAll(const bitCapInt& mask, real1* probsArray)
 
     v = ~mask; // count the number of bits set in v
     bitCapInt skipPower;
-    bitCapInt maxPower = powersVec[powersVec.size() - 1];
     bitLenInt skipLength = 0; // c accumulates the total bits set in v
     std::vector<bitCapInt> skipPowersVec;
     for (skipLength = 0; v; skipLength++) {
         oldV = v;
         v &= v - 1; // clear the least significant bit set
         skipPower = (v ^ oldV) & oldV;
-        if (skipPower < maxPower) {
+        if (skipPower <= mask) {
```

Comment:
That's a bit of a change; intentional?

Reply:
Yes, it is intentional. Note that this is _not_ `maxQPower`. This `maxPower` variable was supposed to be the highest nonzero bit of `mask,` but this is safer.

Reply:
To your point, see https://github.com/vm6502q/qrack/pull/143/commits/37e469a3a31a8fe7c3bab1e3ac4a50f2d8b8aa49. It felt wrong because it actually was wrong, (a little, at least for the OpenCL,) and this is why it became an issue when I tested it on the Raspberry Pi.

The OpenCL kernel ultimately requires an iteration throw every bit in the engine. It wasn't originally clear to me, implementing and debugging, where the iteration to find the powers to skip should stop; it should stop at the last qubit in the engine.